mlir/lib/Dialect/MIOpen/Transforms/AffineTransforms.cpp

```diff
@@ -59,14 +59,62 @@ AffineMap AffineTransforms::buildIndexAffineMap(miopen::TransformOp op) {
-        for (unsigned j = 0; j < srcDimAttr.size(); ++j) {
-          auto srcDim = srcDimAttr.getValue()[j].dyn_cast<IntegerAttr>().getInt();
-          auto destDim = destDimAttr.getValue()[j].dyn_cast<IntegerAttr>().getInt();
-
-          auto expr = getAffineDimExpr(destDim, op.getContext()) + getAffineConstantExpr(-leftPad, op.getContext());
-          affExprsMap.insert({srcDim, expr});
+        // support 2 format of paddding parameters
+        // first is [0, 1] leftpadH = rightPadH = 0, leftpadW = rightPadW = 1
+        if (parameters.size() == 2) {
+          auto leftPad =
+              parameters.getValue()[0].dyn_cast<IntegerAttr>().getInt();
+          auto rightPad =
+              parameters.getValue()[0].dyn_cast<IntegerAttr>().getInt();
+
+          for (unsigned j = 0; j < srcDimAttr.size(); ++j) {
+            auto srcDim =
+                srcDimAttr.getValue()[j].dyn_cast<IntegerAttr>().getInt();
+            auto destDim =
+                destDimAttr.getValue()[j].dyn_cast<IntegerAttr>().getInt();
+
+            auto expr = getAffineDimExpr(destDim, op.getContext()) +
+                        getAffineConstantExpr(-leftPad, op.getContext());
+            affExprsMap.insert({srcDim, expr});
+          }
+        } else { // second is [0, 2, 3, 1] leftpadH = 0 rightPadH =2 leftpadW =
+                 // 3 rightPadW = 1
+          for (unsigned j = 0; j < srcDimAttr.size(); ++j) {
+            auto leftPad =
+                parameters.getValue()[j * 2].dyn_cast<IntegerAttr>().getInt();
+            auto rightPad = parameters.getValue()[j * 2 + 1]
+                                .dyn_cast<IntegerAttr>()
+                                .getInt();
+
+            auto srcDim =
+                srcDimAttr.getValue()[j].dyn_cast<IntegerAttr>().getInt();
+            auto destDim =
+                destDimAttr.getValue()[j].dyn_cast<IntegerAttr>().getInt();
+
+            auto expr = getAffineDimExpr(destDim, op.getContext()) +
+                        getAffineConstantExpr(-leftPad, op.getContext());
+
+            if (leftPad == 0 && rightPad != 0) {
+              auto expr_div = getAffineDimExpr(destDim, op.getContext())
+                                  .ceilDiv(inputShape[srcDim]);
+              // w=14 0-13 right pad =1 ,left pad =0 , the dst is 0-14 <=14 is
+              // the padding block we want to let 0-13 not do transformation,
+              // because dst[0-13]=src[0-13] but if we do not do transformation
+              // ,no affine map generated , and we can't check out of boundary
+              // when miopen-lowering-step4 because of if no affinemap no out of
+              // boundary check, so we will load dst[14], it's not a real memory
+              // address the idea of fixing it is: dst[0-13] not do transform ,
+              // dst[14] do transform with minus src_index = dst_index +
+              // ceil(dst_index/14) -1 , src[0]= dst[0]+1-1=0+1-1=0        <=
```

Comment:
comments here need to be re-formatted. it's hard to see wrapped around math equations.